auth0/src/main/java/com/auth0/android/provider/OAuthManager.java

```diff
@@ -154,13 +156,15 @@ public void onSuccess(@NonNull Credentials codeCredentials) {
 
     //Helper Methods
 
-    private void assertNoError(String errorValue) throws AuthenticationException {
+    private void assertNoError(String errorValue, String errorDescription) throws AuthenticationException {
         if (errorValue == null) {
             return;
         }
         Log.e(TAG, "Error, access denied. Check that the required Permissions are granted and that the Application has this Connection configured in Auth0 Dashboard.");
         if (ERROR_VALUE_ACCESS_DENIED.equalsIgnoreCase(errorValue)) {
             throw new AuthenticationException(ERROR_VALUE_ACCESS_DENIED, "Permissions were not granted. Try again.");
+        } else if (ERROR_VALUE_UNAUTHORIZED.equalsIgnoreCase(errorValue)) {
+            throw new AuthenticationException(ERROR_VALUE_UNAUTHORIZED, errorDescription);
```

Comment:
Yes! If the key doesn't exist on the map, then the returned value will be `null`. This case is covered by the behavior of the `AuthenticationException` class. getDescription will [always](https://github.com/auth0/Auth0.Android/blob/master/auth0/src/main/java/com/auth0/android/authentication/AuthenticationException.java#L131) return a valid message. Maybe instead of checking for null I can use `TextUtils.isEmptyString(description)` [here](https://github.com/auth0/Auth0.Android/blob/master/auth0/src/main/java/com/auth0/android/authentication/AuthenticationException.java#L125) so it checks for null and empty strings. What do you think?